### PR TITLE
[FEATURE] Add SageMaker Pipeline local mode example with BYOC and FrameworkProcessor

### PR DESCRIPTION
**Pull request by brunopistone**
*Description of changes:*
 Publishing example notebook for new Sagemaker Pipeline local mode with FrameworkProcessor for the Processing Step and Custom Pytorch Container with sagemaker-training-toolkit for Training Step

*Testing done:*

 Notebook has been tested locally

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [X] I have read the [CONTRIBUTING](https://github.com/aws/amazon-sagemaker-examples/blob/master/CONTRIBUTING.md) doc and adhered to the example notebook best practices
- [X] I have updated any necessary documentation, including [READMEs](https://github.com/aws/amazon-sagemaker-examples/blob/master/README.md)
- [X] I have tested my notebook(s) and ensured it runs end-to-end
- [X] I have linted my notebook(s) and code using `black-nb -l 100 {path}/{notebook-name}.ipynb`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: sagemaker-pipelines/tabular/local-mode/framework-processor/sagemaker-pipelines-local-mode-fp.ipynb [hunk 1]
```diff
@@ -0,0 +1,945 @@
+{
```
Comment (aaronmarkham) [inline sagemaker-pipelines/tabular/local-mode/framework-processor/sagemaker-pipelines-local-mode-fp.ipynb:1]: <p>Need to make sure this dataset is approved and get it uploaded to the sagemaker-sample-files bucket.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/aws/amazon-sagemaker-examples/pull/3614/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

**Review by aaronmarkham — CHANGES_REQUESTED** [review]
Hi Bruno, 
You've got an empty requirement.txt file in here. Also, not sure we should add your .dockerignore file here.
Can you please get back to me about the dataset... needs to be approved by legal and we should host it in our datasets bucket, so it's replicated regionally.
Some additional narrative would be good in each section of your notebook to describe what's going on.
Thanks!

**Comment by brunopistone** [comment]
> Hi Bruno, You've got an empty requirement.txt file in here. Also, not sure we should add your .dockerignore file here. Can you please get back to me about the dataset... needs to be approved by legal and we should host it in our datasets bucket, so it's replicated regionally. Some additional narrative would be good in each section of your notebook to describe what's going on. Thanks!

Hello @aaronmarkham. The dataset is a public dataset from Kaggle (link: https://www.kaggle.com/datasets/mlg-ulb/creditcardfraud) and it is in the list of pre-approved datasets provided by AWS. I will add the description of the dataset in the notebook. Let me know how it should be hosted in our datasets bucket.
About the requirements.txt, I will remove it from the content (it is an example for the possibility to install additional modules at runtime), and I will add additional descriptions for the different sections

**Comment by brunopistone** [comment]
> Hi Bruno, You've got an empty requirement.txt file in here. Also, not sure we should add your .dockerignore file here. Can you please get back to me about the dataset... needs to be approved by legal and we should host it in our datasets bucket, so it's replicated regionally. Some additional narrative would be good in each section of your notebook to describe what's going on. Thanks!

Hello @aaronmarkham, the dataset was already present in the datasets bucket (under `sagemaker-sample-files/datasets/tabular/mlg-ulb-credit-card/creditcard_csv.csv`).
Added in the notebook the description of the dataset, and for each cell added the description of the step as requested.
The grammar check is failing for some strange requests: the name of the dataset (Dal Pozzolo) and Kaggle seems are not accepted. I think it's a mistake. Can you pleaase check the final content and merge it? Thank you!

**Comment by brunopistone** [comment]
@aaronmarkham can you please check if it can be merged? Thank you!

File: sagemaker-pipelines/tabular/local-mode/framework-processor-byoc/sagemaker-pipelines-local-mode-fp.ipynb [hunk 2]
```diff
@@ -0,0 +1,1017 @@
...
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "## Dataset\n",
+    "\n",
+    "The data set (Credit Card Transactions 2019) was downloaded from [Kaggle](https://www.kaggle.com/datasets/ealtman2019/credit-card-transactions).\n",
```
Comment (aaronmarkham) [inline sagemaker-pipelines/tabular/local-mode/framework-processor-byoc/sagemaker-pipelines-local-mode-fp.ipynb:49]: Still referring to the kaggle dataset

**Pull request merged**